iwyu.cc

```diff
@@ -1654,6 +1641,9 @@ class IwyuBaseAstVisitor : public BaseAstVisitor<Derived> {
     if (!type)
       return;
 
+    if (isa<EnumType>(type))
```

Comment:
You could add a comment here to rationalize why `enum` types are special, particularly since the function comment says "regardless of the type being explicitly written in the source code or not" (line 1647/1634).

I'd suggest something like:
```
    // Enum type uses can be ignored. Their size is known (either implicitly
    // 'int' or explicitly using a C++ 11 enum base). Only if an enum type or
    // its enumerators are explicitly mentioned (in a DeclRefExpr) will they be
    // required by IWYU.
```

Not 100% sure I captured that correctly, you can probably be more precise.

Reply:
Thanks!